(New in 2.7.4) Property 'env' does not exist on type 'ImportMeta'.

### Describe the bug

In v2.7.4, `import.meta.env` doesn't pass `tsc --noEmit` out of the box, due to some typing errors.

v2.7.3 didn't have this issue.



### Reproduction

```
yarn create vite demo --template vanilla-ts
cd demo
yarn
yarn build
```
↑ works.

Then add in `main.ts` :
```js
console.log(import.meta.env.DEV);
```
Then build again:
```
yarn build
```
This time it'll fail with error: 
```
error TS2339: Property 'env' does not exist on type 'ImportMeta'.
```

### System Info

```shell
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz
    Memory: 8.47 GB / 22.95 GB
  Binaries:
    Node: 16.5.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.15 - ~\scoop\apps\yarn\current\Yarn\bin\yarn.CMD
    npm: 7.19.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 96.0.4664.110
    Edge: Spartan (44.19041.1266.0), Chromium (96.0.1054.57)
    Internet Explorer: 11.0.19041.1202
  npmPackages:
    vite: 2.7.4 => 2.7.4
```


### Used Package Manager

yarn

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/vue-next instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/vite/types/importMeta.d.ts
+++ b/packages/vite/types/importMeta.d.ts
@@ -1,10 +1,8 @@
-import type { CustomEventName } from './customEvent'
-import type {
-  ErrorPayload,
-  FullReloadPayload,
-  PrunePayload,
-  UpdatePayload
-} from './hmrPayload'
+// This file is an augmentation to the built-in ImportMeta interface
+// Thus cannot contain any top-level imports
+// <https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation>
+
+/* eslint-disable @typescript-eslint/consistent-type-imports */
 
 interface ImportMeta {
   url: string
@@ -27,15 +25,24 @@ interface ImportMeta {
     invalidate(): void
 
     on: {
-      (event: 'vite:beforeUpdate', cb: (payload: UpdatePayload) => void): void
-      (event: 'vite:beforePrune', cb: (payload: PrunePayload) => void): void
+      (
+        event: 'vite:beforeUpdate',
+        cb: (payload: import('./hmrPayload').UpdatePayload) => void
+      ): void
+      (
+        event: 'vite:beforePrune',
+        cb: (payload: import('./hmrPayload').PrunePayload) => void
+      ): void
       (
         event: 'vite:beforeFullReload',
-        cb: (payload: FullReloadPayload) => void
+        cb: (payload: import('./hmrPayload').FullReloadPayload) => void
       ): void
-      (event: 'vite:error', cb: (payload: ErrorPayload) => void): void
+      (
+        event: 'vite:error',
+        cb: (payload: import('./hmrPayload').ErrorPayload) => void
+      ): void
       <T extends string>(
-        event: CustomEventName<T>,
+        event: import('./customEvent').CustomEventName<T>,
         cb: (data: any) => void
       ): void
     }

